awx/conf/views.py

```diff
@@ -161,40 +163,35 @@ class SettingLoggingTest(GenericAPIView):
     filter_backends = []
 
     def post(self, request, *args, **kwargs):
-        defaults = dict()
-        for key in settings_registry.get_registered_settings(category_slug='logging'):
-            try:
-                defaults[key] = settings_registry.get_setting_field(key).get_default()
-            except serializers.SkipField:
-                defaults[key] = None
-        obj = type('Settings', (object,), defaults)()
-        serializer = self.get_serializer(obj, data=request.data)
-        serializer.is_valid(raise_exception=True)
-        # Special validation specific to logging test.
-        errors = {}
-        for key in ['LOG_AGGREGATOR_TYPE', 'LOG_AGGREGATOR_HOST']:
-            if not request.data.get(key, ''):
-                errors[key] = 'This field is required.'
-        if errors:
-            raise ValidationError(errors)
-
-        if request.data.get('LOG_AGGREGATOR_PASSWORD', '').startswith('$encrypted$'):
-            serializer.validated_data['LOG_AGGREGATOR_PASSWORD'] = getattr(
-                settings, 'LOG_AGGREGATOR_PASSWORD', ''
-            )
-
+        # Error if logging is not enabled
+        enabled = getattr(settings, 'LOG_AGGREGATOR_ENABLED', False)
+        if not enabled:
+            return Response({'error': 'Logging not enabled'}, status=status.HTTP_400_BAD_REQUEST)
+        
+        # Send test message to configured logger based on db settings
+        logging.getLogger('awx').error('AWX Connection Test Message')
```

Comment:
I don't know where would be the right place but I will leave this comment here since it is related to sending the test connection.

When clicking on the test connection and having a http/https server listening to it, the server fails to parse the sent message as JSON because a invalid JSON is being sent:

```
b" '2020-03-25T20:42:01.606Z', 'message': 'AWX Connection Test Message', 'host': 'ip-10-0-1-157.ec2.internal', 'level': 'ERROR', 'logger_name': 'awx', 'stack_info': None, 'cluster_host_id': 'localhost', 'tower_uuid': None}"
```

There is a empty space starting the string sent but no opening `{` for the JSON object. That causes parsing errors. I was also able to observe the same when testing, for example, loggly manually.

Reply:
I observed the same issue with other messages such as the following:

```
b' \'2020-03-25T21:23:58.187Z\', \'message\': "supervisorctl restart finished, stdout:\\nb\'tower-processes:awx-rsyslogd: stopped\\\\ntower-processes:awx-rsyslogd: started\'", \'host\': \'ip-10-0-1-157.ec2.internal\', \'level\': \'INFO\', \'logger_name\': \'awx.main.utils.reload\', \'stack_info\': None, \'cluster_host_id\': \'localhost\', \'tower_uuid\': None}'
```

Is it supposed to be a valid JSON?

Reply:
@elyezer did this get addressed?  If not, it sounds like a blocker to me.

Reply:
I'm looking in to it now

Reply:
Fixed by https://github.com/ansible/awx/pull/6108/commits/e49319e0f238d8cfd4553cd460aae98526ed6ec0.